index.js

```diff
@@ -0,0 +1,98 @@
+'use strict'
+
+const { extractSocketDetails, protocolDecoder } = require('aedes-protocol-decoder')
+const http = require('http')
+const https = require('https')
+const http2 = require('http2')
+const net = require('net')
+const tls = require('tls')
+const WebSocket = require('ws')
+
+const defaultOptions = {
+  ws: null,
+  http: {},
+  https: null,
+  http2: null,
+  tls: null,
+  trustProxy: false,
+  serverFactory: null,
+  protocolDecoder: protocolDecoder,
+  extractSocketDetails: extractSocketDetails
+}
+
+const createServer = (aedes, options) => {
+  if (!aedes || !aedes.handle) {
+    throw new Error('Missing aedes handler')
+  }
+
+  options = Object.assign({}, defaultOptions, options)
+  const aedesHandler = aedes.handle
+
+  let server = null
+  if (options.serverFactory) {
+    server = options.serverFactory(aedes, options)
+  } else if (options.tls) {
+    server = tls.createServer(options.tls, (conn) => {
+      bindConnection(options, aedesHandler, conn)
+    })
+  } else if (options.ws) {
+    if (options.https) {
+      if (options.http2) {
+        server = http2.createSecureServer({ ...options.http2, ...options.https })
+      } else {
+        server = https.createServer({ ...options.http, ...options.https })
+      }
+    } else {
+      if (options.http2) {
+        server = http2.createServer(options.http2)
+      } else {
+        server = http.createServer(options.http)
+      }
+    }
```

Comment:
```suggestion
    } else if (options.http2) {
        server = http2.createServer(options.http2)
    } else if(options.http) {
        server = http.createServer(options.http)
    } else {
        throw Error('No server can be created with options provided')
    }
```

Here too, it's not sure `options.http != null`, another solution could be to automatically set it to `{}` if user set it to `null`?

Reply:
We should allow user to pass null or no http option at all.
So i went for `options.http || {}` same for the merged object when there is options.https.

Reply:
> We should allow user to pass null or no http option at all.

Sure but what if there are no server to create? I mean if options are: `{http: null}` this will throw an error

Reply:
If `ws` is true, (maybe later that will be on object to configure WS.stream), and no `http` options is defined (null or undefined), then we just call http.createServer(options.http). And it works just fine, i did it on `example.js`.